File: tlb/12/solutions/print_posts.pl
#!/usr/bin/perl -w

$dataset = "dataset-medium";

sub student_post_filenames {
	my ($student) = @_;
	
	my @post_filenames = glob "$dataset/$student/*.txt";
	return reverse sort @post_filenames;
}

foreach $student (@ARGV) {
	print join(" ", student_post_filenames($student)), "\n";
}

File: tlb/12/solutions/print_posts.py
#!/usr/bin/python3

import glob, os, sys

dataset = "dataset-medium"

def student_post_filenames(student):
	post_filenames = glob.glob(os.path.join(dataset, student, '*.txt'))
	return sorted(post_filenames, reverse=True) 

for user in sys.argv[1:]:
	print(" ".join(student_post_filenames(user)))
